packages/legend-query/src/stores/QueryBuilderState.ts

```diff
@@ -393,6 +399,19 @@ export class QueryBuilderState {
     }
   }
 
+  changeClass(val: Class | undefined, isRebuildingState?: boolean): void {
+    this.resetQueryBuilder();
+    this.resetQuerySetup();
+    this.querySetupState.setClass(val, isRebuildingState);
+    this.explorerState = new QueryBuilderExplorerState(this);
```

Comment:
Don't think you need this line. Since explorer state was already reset by `resetQuerySetup`